language/Type-Classes.md

```diff
@@ -80,6 +102,63 @@ In fact, a type similar to this `AddInt` is provided in `Data.Monoid.Additive`,
 
 For multi-parameter type classes, the orphan instance check requires that the instance is either in the same module as the class, or the same module as at least one of the types occurring in the instance. (TODO: example)
 
+## Kinds
+
+Type class parameters do not need to be of kind Type. To explicitly specify a different kind, the parameter can be annotated with a kind signature. Unless the kind of a type parameter is explicitly annotated, it will be implicitly inferred based on how the type parameter is used in the type class's method signatures.
+
+``` purescript
+class A a
+-- with a kind signature or methods, `a` defaults to kind `Type`.
+class B (b :: Type)
+-- `b` is specified as kind `Type`, so its instances must also be this kind.
+class C c
+  f :: forall a b. (a -> b) -> c a -> c b
+-- `c`'s kind is unspecified, so it is inferred to be `Type -> Type` by its method `f`.
+class D (d :: Type -> Type)
+-- `D` has no methods, but we can still declare it to apply to types of kind `Type -> Type`.
+class E (e :: # Type)
+-- Row is its own kind, so we can specify a class to operate on types of rows, `# Type`.
+```
+
+The kinds of any types mentioned in an instance declaration must match the kinds of those parameters as determined by the type class's definition.
+
+For example:
+
+``` purescript
+-- Functor can only be considered for types of kind `Type -> Type`.
+class Functor f where
+  map :: forall a b. (a -> b) -> f a -> f b
+-- Either is kind `Type -> Type -> Type`.
+data Either a b = Left a | Right b
+-- This means we can't write an instance of Functor for Either.
+`instance functorEither :: Functor Either where ...` -- compile error
+-- To make Either into kind `Type -> Type`, we simply include the first type parameter.
+`instance functorEither :: Functor (Either e) where ...`
+```
+
+An effect of needing to adjust the kind of a data type to fit into a type class which applies to a lower arity kind, like with `Functor` and `Either` above, is that the behavior and laws of an instance will bias towards to the right-most type parameters.
```

Comment:
I'm not sure about this sentence; I think it will only make sense to people who already understand what it's trying to explain. This topic is definitely worth addressing but I think we should do that separately from this PR.